### PR TITLE
Cr 2468 not car azure ensure postgre sql servers have connection throttling enabled + CR-2469 Ensure PostgreSQL Servers have log checkpoints enabled

### PR DESCRIPTION
**Pull request by lirshindalman**

### CODE REVIEW

File: cloudrail/knowledge/context/azure/resources/databases/azure_postgresql_server.py [hunk 1]
```diff
@@ -40,5 +129,5 @@ def is_tagable(self) -> bool:
         return True
 
     def to_drift_detection_object(self) -> dict:
-        return {'tags': self.tags, 'server_name': self.server_name,
+        return {'name': self.name,
```
Comment (Imanuel-Mizrahi) [inline cloudrail/knowledge/context/azure/resources/databases/azure_postgresql_server.py:132]: I think that the "name" should not be inside the drift detection object, as we cannot change it, without creating a new resource, and it's also being used in the "get_keys" (this is not your mistake, but I think we need to remove it now)
We need to add the "tags" again.
Probably now that we have many new fields, we should add it as well.

**Review by Imanuel-Mizrahi — CHANGES_REQUESTED** [review]

**Review by Imanuel-Mizrahi — APPROVED** [review]

File: cloudrail/knowledge/context/azure/azure_relations_assigner.py [hunk 2]
```diff
@@ -102,6 +107,15 @@ def _assign_config_to_function_app(function_app: AzureFunctionApp, app_service_c
         )
         function_app.app_service_config = app_service_config
 
+    @staticmethod
+    def _assign_config_to_postgresql_server(postgresql_server: AzurePostgreSqlServer,
+                                            postgresql_server_configs: AliasesDict[AzurePostgreSqlServerConfiguration]):
+        postgresql_server.postgresql_configuration = ResourceInvalidator.get_by_logic(
```
Comment (Tomerva) [inline cloudrail/knowledge/context/azure/azure_relations_assigner.py:113]: (last pr comment)
is it possible that psql doesnt have config? if not, it should be invalidated

File: cloudrail/knowledge/context/azure/resources/databases/azure_postgresql_server.py (hunk 1 above)
Comment (Tomerva) [reply cloudrail/knowledge/context/azure/resources/databases/azure_postgresql_server.py:132]: 1. add tags
2. get_keys should include subscription id

File: cloudrail/knowledge/rules/azure/non_context_aware/abstract_postgresql_servers_have_configuration_value_enabled_rule.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        pass
+
+    @abstractmethod
+    def get_configuration_type(self):
+        pass
+
+    def execute(self, env_context: AzureEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for postgresql_servers in env_context.postgresql_servers:
+            if postgresql_servers.postgresql_configuration.name == self.get_configuration_type() and\
```
Comment (Tomerva) [inline cloudrail/knowledge/rules/azure/non_context_aware/abstract_postgresql_servers_have_configuration_value_enabled_rule.py:27]: so i assume config is mandatory?..

File: tests/knowledge/context/azure/test_postgresql_server.py [hunk 4]
```diff
@@ -20,3 +21,32 @@ def test_postgresql_enforcing_ssl_not_enabled(self, ctx: AzureEnvironmentContext
         server = ctx.postgresql_servers.get('cr2467-postgresql-server')
         self.assertIsNotNone(server)
         self.assertFalse(server.ssl_enforcement_enabled)
+
+    @context(module_path="basic", test_options=TestOptions(run_drift_detection=False, run_cloudmapper=False))
```
Comment (Tomerva) [inline tests/knowledge/context/azure/test_postgresql_server.py:25]: remove test options?

File: tests/knowledge/context/azure/test_postgresql_server_configuration.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+from tests.knowledge.context.azure_context_test import AzureContextTest
+from tests.knowledge.context.test_context_annotation import context, TestOptions
+
+
+class TestAzurePostgreSqlServerConfiguration(AzureContextTest):
+
+    def get_component(self):
+        return "postgresql_server_configuration"
+
+    @context(module_path="basic", test_options=TestOptions(run_drift_detection=False))
```
Comment (Tomerva) [inline tests/knowledge/context/azure/test_postgresql_server_configuration.py:12]: remove test options?

**Review by Tomerva — CHANGES_REQUESTED** [review]

File: cloudrail/knowledge/rules/azure/non_context_aware/abstract_postgresql_servers_have_configuration_value_enabled_rule.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+    @abstractmethod
+    def get_configuration_type(self):
+        pass
+
+    def execute(self, env_context: AzureEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for postgresql_servers in env_context.postgresql_servers:
+            if postgresql_servers.postgresql_configuration:
+                if postgresql_servers.postgresql_configuration.name == self.get_configuration_type() and\
```
Comment (Tomerva) [inline cloudrail/knowledge/rules/azure/non_context_aware/abstract_postgresql_servers_have_configuration_value_enabled_rule.py:28]: im a little confused on why its called name but is checked against type

**Review by Tomerva — CHANGES_REQUESTED** [review]

**Review by Tomerva — APPROVED** [review]

**Pull request merged**